pkg/scheduler/framework/plugins/tainttoleration/filtering_test.go

```diff
@@ -0,0 +1,446 @@
+					},
+				},
+			},
+			wantStatus: framework.NewNonErrorStatus(framework.ClusterUnschedulable, p.Name(), fmt.Sprintf(reasonFmt, &clusterv1beta1.Taint{Key: "key1", Value: "value1", Effect: corev1.TaintEffectNoSchedule})),
+		},
+		{
+			name: "taints can be tolerated based on key, value & effect with Equal operator, every taint has matching toleration - nil status",
+			cluster: &clusterv1beta1.MemberCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-mc",
+				},
+				Spec: clusterv1beta1.MemberClusterSpec{
+					Taints: []clusterv1beta1.Taint{
+						{
+							Key:    "key1",
+							Value:  "value1",
+							Effect: corev1.TaintEffectNoSchedule,
+						},
+						{
+							Key:    "key2",
+							Value:  "value2",
+							Effect: corev1.TaintEffectNoSchedule,
+						},
+					},
+				},
+			},
+			policySnapshot: &placementv1beta1.ClusterSchedulingPolicySnapshot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "csp-1",
+				},
+				Spec: placementv1beta1.SchedulingPolicySnapshotSpec{
+					Policy: &placementv1beta1.PlacementPolicy{
+						PlacementType: placementv1beta1.PickAllPlacementType,
+						Tolerations: []placementv1beta1.Toleration{
+							{
+								Key:      "key1",
+								Operator: corev1.TolerationOpEqual,
+								Value:    "value1",
+								Effect:   corev1.TaintEffectNoSchedule,
+							},
+							{
+								Key:      "key2",
+								Operator: corev1.TolerationOpEqual,
+								Value:    "value2",
+								Effect:   corev1.TaintEffectNoSchedule,
+							},
+						},
+					},
+				},
+			},
+			wantStatus: nil,
+		},
+		{
+			name: "taints can be tolerated based on key, value & empty effect with Equal operator, every taint has matching toleration - nil status",
+			cluster: &clusterv1beta1.MemberCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-mc",
+				},
+				Spec: clusterv1beta1.MemberClusterSpec{
+					Taints: []clusterv1beta1.Taint{
+						{
+							Key:    "key1",
+							Value:  "value1",
+							Effect: corev1.TaintEffectNoSchedule,
+						},
+						{
+							Key:    "key2",
+							Value:  "value2",
+							Effect: corev1.TaintEffectNoSchedule,
+						},
+					},
+				},
+			},
+			policySnapshot: &placementv1beta1.ClusterSchedulingPolicySnapshot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "csp-1",
+				},
+				Spec: placementv1beta1.SchedulingPolicySnapshotSpec{
+					Policy: &placementv1beta1.PlacementPolicy{
+						PlacementType: placementv1beta1.PickAllPlacementType,
+						Tolerations: []placementv1beta1.Toleration{
+							{
+								Key:      "key1",
+								Operator: corev1.TolerationOpEqual,
+								Value:    "value1",
+							},
+							{
+								Key:      "key2",
+								Operator: corev1.TolerationOpEqual,
+								Value:    "value2",
+							},
+						},
+					},
+				},
+			},
+			wantStatus: nil,
+		},
+		{
+			name: "taints can be tolerated based on key, empty value, effect with Equal operator, every taint has matching toleration - nil status",
```

Comment:
add a negative test case for taints (key present + empty value + operation is equal)?